### PR TITLE
Modernize NoOp CipherExecutor factory API

### PR DESCRIPTION
**Pull request by dima767**
In particular, hide the concrete `NoOpCipherExecutor` class behind package-private access level and introduce strongly typed factory methods for it in `CipherExecutor` interface itself, taking advantage of Java8+ interface static methods.

If the question arises `what is the advantage of this?` - it's just cleaner API with clear encapsulation following modern API design patterns. That is all ;-)

This pattern of hiding concrete implementation classes and factory methods on interfaces could be applied throughout CAS code base where appropriate.

### CODE REVIEW

File: api/cas-server-core-api-util/src/main/java/org/apereo/cas/NoOpCipherExecutor.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    /**
+     * Gets instance.
+     *
+     * @return the instance
+     */
+    @SuppressWarnings("unchecked")
+    static <I, O> CipherExecutor<I, O> getInstance() {
+        //Double-check pattern here to ensure correctness of only single instance creation in multi-threaded environments
+        if (INSTANCE == null) {
```
[the remark this answers is not in the document]
Comment (dima767) [reply api/cas-server-core-api-util/src/main/java/org/apereo/cas/NoOpCipherExecutor.java]: It is thread safe. That's the point of double-check pattern. Bogus "issue"

File: api/cas-server-core-api-util/src/main/java/org/apereo/cas/NoOpCipherExecutor.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * @author Misagh Moayyed
+ * @author Dmitriy Kopylenko
+ *
+ * @since 4.1
+ */
+@Slf4j
+@Getter
+class NoOpCipherExecutor<I, O> implements CipherExecutor<I, O> {
+
+    private static volatile CipherExecutor INSTANCE;
```
[the remark this answers is not in the document]
Comment (dima767) [reply api/cas-server-core-api-util/src/main/java/org/apereo/cas/NoOpCipherExecutor.java]: Bogus. It is singleton

File: api/cas-server-core-api-util/src/main/java/org/apereo/cas/NoOpCipherExecutor.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    /**
+     * Gets instance.
+     *
+     * @return the instance
+     */
+    @SuppressWarnings("unchecked")
```
[the remark this answers is not in the document]
Comment (dima767) [reply api/cas-server-core-api-util/src/main/java/org/apereo/cas/NoOpCipherExecutor.java]: Bogus. The whole idea is to hide this class in package level scope

**Comment by mmoayyed** [comment]
This is perfectly good with me. Thanks Dima.

**Pull request merged**